packages/auth-sveltekit/src/server.ts

```diff
@@ -0,0 +1,489 @@
+import type { Client } from "edgedb";
+import {
+  Auth,
+  builtinOAuthProviderNames,
+  type BuiltinOAuthProviderNames,
+  type TokenData,
+  type emailPasswordProviderName,
+} from "@edgedb/auth-core";
+import {
+  ClientAuth,
+  getConfig,
+  type AuthConfig,
+  type AuthOptions,
+} from "./client.js";
+
+export type { TokenData, AuthOptions, Client };
+
+export type BuiltinProviderNames =
+  | BuiltinOAuthProviderNames
+  | typeof emailPasswordProviderName;
+
+type ParamsOrError<Result extends object, ErrorDetails extends object = {}> =
+  | ({ error: null } & { [Key in keyof ErrorDetails]?: undefined } & Result)
+  | ({ error: Error } & ErrorDetails & { [Key in keyof Result]?: undefined });
+
+export interface AuthRouteHandlers {
+  onOAuthCallback(
+    params: ParamsOrError<{
+      tokenData: TokenData;
+      provider: BuiltinOAuthProviderNames;
+      isSignUp: boolean;
+    }>
+  ): Promise<Response>;
+  onBuiltinUICallback(
+    params: ParamsOrError<
+      (
+        | {
+            tokenData: TokenData;
+            provider: BuiltinProviderNames;
+          }
+        | { tokenData: null; provider: null }
+      ) & { isSignUp: boolean }
+    >
+  ): Promise<Response>;
+  onEmailVerify(
+    params: ParamsOrError<
+      { tokenData: TokenData },
+      { verificationToken?: string }
+    >
+  ): Promise<Response>;
+  onSignout(): Promise<Response>;
+}
+
+export default function serverAuth(client: Client, options: AuthOptions) {
+  const core = Auth.create(client);
+  const config = getConfig(options);
+
+  return {
+    createServerRequestAuth: ({ event }: { event: RequestEvent }) =>
+      new ServerRequestAuth(client, core, event, options),
+    createAuthRouteHook:
+      (handlers: AuthRouteHandlers): Handle =>
+      ({ event, resolve }) => {
+        const pathname = new URL(event.request.url).pathname;
+
+        if (pathname.startsWith(`/${config.authRoutesPath}`)) {
+          return createAuthRouteHandlers(handlers, event, core, config);
+        }
+
+        return resolve(event);
+      },
+  };
+}
+
+export class ServerRequestAuth extends ClientAuth {
+  private readonly client: Client;
+  private readonly core: Promise<Auth>;
+  private readonly cookies: Cookies;
+
+  public get session() {
+    return new AuthSession(
+      this.client,
+      this.cookies.get(this.config.authCookieName)
+    );
+  }
+
+  /** @internal */
+  constructor(
+    client: Client,
+    core: Promise<Auth>,
+    { cookies }: RequestEvent,
+    options: AuthOptions
+  ) {
+    super(options);
+
+    this.client = client;
+    this.core = core;
+    this.cookies = cookies;
+  }
+
+  isPasswordResetTokenValid(resetToken: string) {
+    return Auth.checkPasswordResetTokenValid(resetToken);
+  }
+
+  async getProvidersInfo() {
+    return (await this.core).getProvidersInfo();
+  }
+
+  async emailPasswordSignUp({
+    email,
+    password,
+  }: {
+    email: string;
+    password: string;
+  }): Promise<{ tokenData?: TokenData | null }> {
```

Comment:
Did we intentionally drop the `{email: string, password: string} | FormData` thing, where you can just pass the formData object directly to these methods?

Reply:
I guess I removed FormData support because supabase and lucia didnt have it too, added it back.